terminal/src/identity.rs

```diff
@@ -0,0 +1,63 @@
+use librad::profile::Profile;
+
+use radicle_common::fmt;
+use radicle_common::profile;
+
+use crate as term;
+
+/// Identity formatter that takes a profile and displays it `peer_id` and
+/// `username` depending on the configuration.
+///
+/// If short display is requested, `peer_id` is printed in its compact format
+/// e.g. `hynddpk…uf4qwge`. If `styled` is requested, `peer_id` and `username`
+/// are printed using the terminal's styled formatters.
+pub struct Formatter<'a> {
+    profile: &'a Profile,
+    shorten: bool,
+    style: bool,
+}
```

Comment:
This is a nice idea!

Reply:
Glad you like it! I introduced this because I think in the future we could potentially share formatting code with the TUI, which works on https://docs.rs/tui/0.19.0/tui/text/struct.Spans.html.

I'd want to see how we could extract shareable components that are independent from how they're written into the buffer.